feature/gribi/otg_tests/static_lsp/static_lsp_test.go

```diff
@@ -0,0 +1,367 @@
+	dstMac := gnmi.Get(t, dut, gnmi.OC().Interface(dutDstInterface).Ethernet().MacAddress().State())
+	t.Logf("DUT remote mac address is %s", dstMac)
+
+	// Create a traffic flow with MPLS
+	flowName := fmt.Sprintf("MPLS-%d-MPLS-%d", mplsLabel1, mplsLabel2)
+	mplsFlow := top.Flows().Add().SetName(flowName)
+	mplsFlow.TxRx().Port().
+		SetTxName(ate.Port(t, "port1").ID()).
+		SetRxNames([]string{ate.Port(t, "port2").ID()})
+
+	mplsFlow.Metrics().SetEnable(true)
+	mplsFlow.Rate().SetPps(500)
+	mplsFlow.Size().SetFixed(512)
+	// mplsFlow.Duration().FixedPackets().SetPackets(100000)
+	mplsFlow.Duration().Continuous()
+
+	// Set up ethernet layer.
+	eth := mplsFlow.Packet().Add().Ethernet()
+	eth.Src().SetValue(ateSrc.MAC)
+	eth.Dst().SetValue(dstMac)
+
+	// Set up MPLS layer with destination label 100.
+	mpls := mplsFlow.Packet().Add().Mpls()
+	mpls.Label().SetValue(label1)
+	mpls.BottomOfStack().SetValue(0)
+
+	// Set up MPLS layer with destination label 100.
+	mpls2 := mplsFlow.Packet().Add().Mpls()
+	mpls2.Label().SetValue(label2)
+	mpls2.BottomOfStack().SetValue(1)
+
+	ip4 := mplsFlow.Packet().Add().Ipv4()
+	ip4.Src().SetValue(ateSrc.IPv4)
+	ip4.Dst().SetValue(ateDst.IPv4)
+	ip4.Version().SetValue(4)
+
+	return mplsFlow
+
+}
+
+func ValidatePackets(t *testing.T, filename string, expextedLabel uint32) {
+	handle, err := pcap.OpenOffline(filename)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer handle.Close()
+
+	// Define tolerance
+	const tolerancePercentage = 1.0
+	var expectedMPLSPackets int
+	var unexpectedMPLSPackets int
+
+	// Convert string to net.IP for comparison
+	targetSrcIP := net.ParseIP(ateSrc.IPv4)
+	targetDstIP := net.ParseIP(ateDst.IPv4)
+
+	t.Logf("Checking Packets to verify MPLS label was popped and searching for "+
+		"src %s and dst %s", ateSrc.IPv4, ateDst.IPv4)
+
+	packetSource := gopacket.NewPacketSource(handle, handle.LinkType())
+	for packet := range packetSource.Packets() {
+		ipLayer := packet.Layer(layers.LayerTypeIPv4)
+		if ipLayer == nil {
+			t.Log("IP layer not found in packet")
+			continue
+		}
+		ipv4, _ := ipLayer.(*layers.IPv4)
+
+		// Compare the source IP address in the packet to the target source IP
+		if ipv4.SrcIP.Equal(targetSrcIP) && ipv4.DstIP.Equal(targetDstIP) {
+			// Now check for an MPLS layer
+			var mpls *layers.MPLS
+			mplsLayer := packet.Layer(layers.LayerTypeMPLS)
+			if mplsLayer != nil {
+				mpls, _ := mplsLayer.(*layers.MPLS)
+				// check if the expected label is found
+				if mpls.Label == expextedLabel {
+					expectedMPLSPackets++
+				} else {
+					t.Errorf("Unexpected Label Found MPLS packet with label: %v", mpls.Label)
+					unexpectedMPLSPackets++
+				}
+			} else {
+				// increment the unexpected packet counter
+				unexpectedMPLSPackets++
+				t.Errorf("Found MPLS packet with label: %v", mpls.Label)
+
+			}
+		}
+	}
+
+	// Calculate the tolerance based on the number of expected MPLS packets
+	tolerance := int(float64(expectedMPLSPackets) * (tolerancePercentage / 100))
```

Comment:
This var name (`tolerance`) shadows the const with the same name from the global block